packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -16,7 +16,9 @@ export const samplerOperations = {
                     .getABIEncodedTransactionData();
             },
             handleCallResultsAsync: async (contract, callResults) => {
-                return contract.getABIDecodedReturnData<BigNumber[]>('getOrderFillableTakerAssetAmounts', callResults);
+                return contract
+                    .getABIDecodedReturnData<BigNumber[]>('getOrderFillableTakerAssetAmounts', callResults)
+                    .map((amt, i) => (orders[i].takerAddress === NULL_ADDRESS ? amt : orders[i].takerAssetAmount));
```

Comment:
Can you please add a comment in the code base explaining what's going on here?

Reply:
I got rid of this commit.  Please see my explanation here: https://github.com/0xProject/0x-monorepo/pull/2555#discussion_r410040348